app/controllers/case_contacts_controller.rb

```diff
@@ -35,18 +35,8 @@ def new
     store_referring_location
     authorize CaseContact
 
-    # - If there are cases defined in the params, select those cases (often coming from the case page)
-    # - If there is only one case, select that case
-    # - If there are no hints, let them select their case
     casa_cases = policy_scope(current_organization.casa_cases)
-    draft_case_ids =
-      if params.dig(:case_contact, :casa_case_id).present?
-        casa_cases.where(id: params.dig(:case_contact, :casa_case_id)).pluck(:id)
-      elsif casa_cases.count == 1
-        [casa_cases.first.id]
-      else
-        []
-      end
+    draft_case_ids = set_draft_case_ids(params, casa_cases)
```

Comment:
Nit: `build_draft_case_ids` maybe? `set` sounds like it is doing the setting via a side effect.

Reply:
agree, can change